### ISSUE TITLE
AAD POD Identity - FailedIdentityOperation while upgrading an AKS cluster

### ISSUE THREAD

**Issue by axelgMS** [Azure/aad-pod-identity#676]
**Describe the bug**

vmss.patch.getResult: resourceID: /subscriptions/<SubID>/resourceGroups/<RGName>/providers/Microsoft.Compute/virtualMachineScaleSets/<VMSSName>, parameters: XYZ, error: autorest/azure: Service returned an error. Status=429 Code="FailedIdentityOperation" Message="Identity operation for resource '/subscriptions/<SubID>/resourceGroups/<RGName>/providers/Microsoft.Compute/virtualMachineScaleSets/<VMSSName>' failed with error 'Failed to perform resource identity operation. Status: '429'. Response: '{\"error\":{\"code\":\"429\",\"message\":\"Request is temporarily throttled because resource /subscriptions/<SubID>/resourceGroups/<RGName>/providers/Microsoft.Compute/virtualMachineScaleSets/<VMSSName> has issued too many requests. Retry after 20 seconds.\"}}'.'."

**Steps To Reproduce**

Unknown - creating this issue on behalf of customer.
What we're seeing is that this customer can't upgrade his AKS Cluster as it's facing throttling. lot of VMSS PATCH operations that are getting 429 conflict errors. They are coming from UserAgent:
    **Go/go1.14.3 (amd64-linux) go-autorest/v14.0.0 Azure-SDK-For-Go/v40.4.0 compute/2019-12-01 aad-pod-identity/MIC/1.6.1/fd105c7/2020-05-22-17:50**

There were more than 150 failed operations over a period of 1.5 hours - and almost 4000 PATCH requests.

**Expected behavior**

no 429 and no FailedIdentityOperation error.
So basically less VMSS PATCH operations triggered from that AAD POD Identity.

**AAD Pod Identity version**

aad-pod-identity-mic:1.6.1

**Kubernetes version**

Azure AKS 1.15.11

**Additional context**

CRI 195602857. Ping me for more details (axelg)

**Comment by aramase** [comment]
@axelg If the cluster is throttled, then it's possible `aad-pod-identity` patch operations fail because of 429 rate limit errors. If there is a `retry-after-header` in the response, then we should honor that in `aad-pod-identity` or set an internal backoff if the error is a 429. I'll open an issue for adding that feature to prevent further throttling if the cluster is already throttled. 

For the current user facing the issue, they should be able to scale down the MIC deployment to 0 replicas until they are no longer throttled and then scale it back up again.

**Comment by aramase** [comment]
Opened #677 to track changes for honor retry-after header or backoff in case of throttling. Closing this issue. Please feel free to reopen if there are any other issues.
